eZ/Publish/Core/Repository/TrashService.php

```diff
@@ -372,4 +380,46 @@ protected function getDateTime($timestamp)
 
         return $dateTime;
     }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location
+     *
+     * @return bool
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    private function userHasPermissionsToRemove(APIContentInfo $contentInfo, Location $location): bool
+    {
+        if (!$this->repository->canUser('content', 'remove', $contentInfo, [$location])) {
+            return false;
+        }
+
+        $contentRemoveCriterion = $this->permissionCriterionResolver->getPermissionsCriterion('content', 'remove');
```

Comment:
Is there any scenario that this can return `true`?

Reply:
accoriding to docs, yes https://github.com/ezsystems/ezpublish-kernel/blob/ae9dff7736e6fcde030278d44ed54a696d35e581/eZ/Publish/API/Repository/PermissionCriterionResolver.php#L16-L26

Reply:
And that was my point ;)